### DISCUSSION TITLE
Custom fields in the backend

### DISCUSSION THREAD

**Discussion by wmarie-joseph** [oliverklee/ext-seminars#2404] [Q&A] [answered]
TYPO3 version : 11.5.22
Seminars version : 5.1.0

Hello,

I was wondering if it was possible to add custom fields in the back-end for seminars ?

Without any risk of these fields being overwriting with each upcoming upgrades to the extension.

Thanks in advance and best regards

**Comment by oliverklee** [comment, marked as answer]
@wmarie-joseph  Yes, you can create an extension that adds the SQL and TCA for these fields, and XCLASSes the corresponding domain model class. If you want to display the corresponding data in the FE single view, you'll need to use a custom HTML template and use the `SeminarSingleView` hook to provide the additional data to the template.

**Comment by haythemys** [comment]
Hello @oliverklee  , i created a custom extension with Extension Builder , and i want to extend the seminars extension using the method you showed above . Is there somebody who did the same thing , if so could you link me to the github ? Best regards

**Reply by oliverklee** [reply]
I don't know of any examples I could point to.

(Also, I'm not sure if this is possible with the extension builder.)

**Reply by haythemys** [reply]
Hello again @oliverklee  , thank you for your quick answer . Can you tell me  please why is not possible with the extension builder ? Best ragards

**Reply by oliverklee** [reply]
Last time I checked (which I have to say was several years ago), the extension builder only supported creating new tables/models, but not extending existing tables. Extending an existing table on the other hand requires the TCA to have a different structure and be in a different folder (`Configuration/TCA/Overrides/`), and the model needs to extend the model it XCLASSes.

**Reply by haythemys** [reply]
Hello , with the new version of Extension Builder , we can extend a extension that exists.
https://docs.typo3.org/p/friendsoftypo3/extension-builder/11.0/en-us/InDepth/ExtendingDomainObjects.html.
 Tried to add another tab on Seminars content  like organizer.
<img width="1408" alt="Capture d’écran 2023-02-16 à 12 26 08" src="https://user-images.githubusercontent.com/85106939/219352713-7ec632ce-2985-4580-b9c8-75b7f64ed333.png">

Can you tell me please which class i can extend to add another tab ? 

Best Regards

**Reply by oliverklee** [reply]
Tabs are defined in the TCA, not in classes.

**Comment by oliverklee** [comment]
Locking this discussion as this question is not specific to the seminars extension, but generally about how to extend tables of extensions. For general support and discussions about TYPO3 that are not specific to the seminars extensions, I recommend using the TYPO3 Slack.
